src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -501,6 +501,19 @@ case class Between(startInterval: Interval, endInterval: Interval) extends Inter
 }
 
 /**
+  * Finds the interval between two input intervals including both intervals. Formally:
+  * Between([t1,t2): startInterval,[t3,t4): endInterval): Interval = [t1,t4)
+  *
+  * @param startInterval first interval
+  * @param endInterval   second interval
+  */
+case class BetweenInclusive(startInterval: Interval, endInterval: Interval) extends Interval {
```

Comment:
I think we may need to add extra parameters to `Between` rather than adding a new class. See the comment above: the start and end can independently be set as inclusive or not.